### PR TITLE
MODIS Sensor Notebook

### PR DESCRIPTION
**Pull request by samueljackson92**
A PR for contributing a MODIS data demonstrator notebook to the Environmental AI book. This notebook:

* Use `satpy` to load, visualise, and regrid MODIS level 1B data.
* Fetch a fire database containing some 497364 fires from 2020.
* Visualisation of fire pixels from the database
* Visualisation of the fire pixels alongside bands from the MODIS satellite data.

A couple of notes on information sources:
 - The MODIS satellite products are available via stac, BUT they are exposed as RasterIO's to intake, which cannot (yet) be read from a remote source by `intake_xarray`. So instead I just download the file.
 - The fire data is publicly accessible via a zip folder. So for now I again just download that and unzip it. 

@acocac let me know if there is anything I need to adjust or add.

### CODE REVIEW

**Comment by acocac** [comment]
Hey @samueljackson92  - the use case looks great! I've made a few suggestions. Here are some additional thoughts:
- For **Fetch data**, we should add a **warning** to report users must enable the **ladsweb** service in their Earth Data account. 
- We will still need to **improve the description** of certain subheadings in the simplest terms and where possible avoid jargon. For instance, in the last cell, you overlay FRP points and MODIS Band 20? Why that? Is it a thermal band? 
- I would suggest closing with a **Summary section**. This should have a quick recap of the main aspects, contributions and technologies (libraries) shown in the notebook.

I hope the above suggestions are useful. Keep going :fire:

**Comment by samueljackson92** [comment]
@acocac Thanks for the feedback. I agree with all the above. I shall have a go at making the changes above and come back to you!

**Comment by samueljackson92** [comment]
Hi @acocac,

Sorry for the long silence. I've updated the notebook to:

- Added a noticed to authorise your earthdata account before downloading data.
- Added a check which will notify you if you could not download data because of account/permission issues.
- Added a short summary section
- Added more descriptions to different sections of the notebook
- Reduced the number of fires visualised, so that it's not quite so heavy weight.

Let me know if you spot anything else or if I missed something.

File: book/wildfires/sensors/wildfires-sensors-modis.ipynb [hunk 1]
```diff
@@ -0,0 +1,589 @@
+{
```
Comment (acocac) [inline book/wildfires/sensors/wildfires-sensors-modis.ipynb:1]: <p><code>pyspectral</code> isn't in the default environment so we should install it first:</p><pre class="ql-syntax" spellcheck="false">!pip -q install pyspectral
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/acocac/environmental-ai-book/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (acocac) [inline book/wildfires/sensors/wildfires-sensors-modis.ipynb:1]: <p>add <code>import urllib.request</code> if we use <code>urllib.request.urlretrieve</code> to download MODIS wildfires, see comment in the line<code>! wget 'https://firms2.modaps.eosdis.nasa.gov/data/country/zips/modis_2020_all_countries.zip' -O 'modis_2020_all_countries.zip'</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/acocac/environmental-ai-book/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (acocac) [inline book/wildfires/sensors/wildfires-sensors-modis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    ! wget 'https://firms2.modaps.eosdis.nasa.gov/data/country/zips/modis_2020_all_countries.zip' -O 'modis_2020_all_countries.zip'
</pre><p> </p><p>I would suggest to replace by <code>urllib.request.urlretrieve </code>as wget isn't installed by default in all OS.</p><p> </p><pre class="ql-syntax" spellcheck="false">filepath = 'https://firms2.modaps.eosdis.nasa.gov/data/country/zips/'
filename = 'modis_2020_all_countries.zip'
urllib.request.urlretrieve(filepath+filename, '../sensors/' + filename)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/acocac/environmental-ai-book/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

Comment (acocac) [inline book/wildfires/sensors/wildfires-sensors-modis.ipynb:1]: <p>When referring to libraries you can highlight them <code>hvplot</code> in the text.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/acocac/environmental-ai-book/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='23'/>

**Comment by acocac** [comment]
Hi @samueljackson92, LGTM! 
I added some minor changes. See them via [ReviewNB](https://app.reviewnb.com/acocac/environmental-ai-book/pull/2/discussion/). If they're ok to you, then I can merge and test the notebook runs in`Binder`.

**Comment by samueljackson92** [comment]
Hi @acocac, I believe I've made all the requested changes. Let me know how the Binder testing goes?

**Comment by acocac** [comment]
Sam, LGTM! I'll close the issue once it works in the Binder. Thanks!!!

**Comment by samueljackson92** [comment]
@acocac Great! 

I will also try and take a look at making one for the SEVIRI sensor at some point soon!

**Comment by acocac** [comment]
@samueljackson92 
I confirm the notebook works now in Binder. I made some minor modifications to get a rendered version. Thanks to your use case, I also learnt that declaring `rasterize=True` in `hvplot.image` considerably reduces the notebook size. 
Looking forward to assist on the new one for the SEVIRI sensor.
Alejandro

**Pull request merged**